components/basecomponents/Button.tsx

```diff
@@ -1,14 +1,39 @@
 import { forwardRef } from 'react';
+import PlayIcon from 'public/icons/playicon.svg';
+import PlaySlowIcon from 'public/icons/slowplay.svg';
 
-interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
-  text: string;
+const BASE_STYLE = 'rounded-lg';
+const TEXT_PADDING = 'px-5 py-2';
+const ICON_PADDING = 'px-2 py-2';
+
+const BUTTON_STYLE = {
+  default: `${BASE_STYLE} text-black bg-primary-grey`,
+  primary: `${BASE_STYLE} text-white bg-primary-blue disabled:grayscale disabled:opacity-75`,
+  primaryLight: `${BASE_STYLE} text-black bg-primary-yellow`,
+  choice: `${BASE_STYLE} text-black border border-slate-300 hover:border-slate-400 bg-white`,
+  choiceCorrect: `${BASE_STYLE} text-black border border-slate-300 hover:border-slate-400 bg-primary-green`,
+};
+
+interface ButtonProps extends React.ComponentProps<'button'> {
+  buttonStyle?: keyof typeof BUTTON_STYLE;
+  icon?: boolean | 'play' | 'playSlow';
+  shadow?: boolean;
 }
 
-export const Button = forwardRef((props: ButtonProps, ref: React.ForwardedRef<HTMLButtonElement>): JSX.Element => {
-  const { text, className, ...rest } = props;
-  return (
-    <button className={`${className}  text-white  h-auto rounded-lg py-2 px-5`} ref={ref} {...rest}>
-      {text}
-    </button>
-  );
-});
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ children, className, buttonStyle = 'default', icon = false, shadow = false, ...rest }, ref) => {
+    return (
+      <button
+        ref={ref}
+        className={`tap-transparent ${className} ${BUTTON_STYLE[buttonStyle]} ${icon !== false ? ICON_PADDING : TEXT_PADDING} ${
+          shadow ? 'button-shadow' : ''
+        }`}
+        {...rest}
+      >
+        {icon === 'play' && <PlayIcon className="inline h-auto" />}
```

Comment:
Instead of these conditional renderings, you should have an `icon` prop accepting a JSX element and pass whatever icon you want :)

Reply:
It's meant to use icon=true for proper padding and pass icon component as children for custom icon. Use of icon=play is a bonus :-) to have button with play icon out of the box with same icon through whole site.

Reply:
"to have button with play icon out of the box with same icon through whole site" Then I think it should be a separate component that uses <Button> with the Play icon inside `props.children`.

Reply:
Could be.